test/test_actor.py

```diff
@@ -107,3 +113,61 @@ def test_rotation(self):
         for _ in range(360):
             a.angle += 1.0
         self.assertEqual(a.pos, (100.0, 100.0))
+
+    def test_scaling(self):
+        actor = Actor('alien')
+        originial_size = actor.size
+
+        # No scaling
+        actor.scale = (1, 1)
+        self.assertEqual(actor.size, originial_size)
+
+        # Scaling on x-axis only
+        actor.scale_x = 2
+        self.assertEqual(actor.size, (originial_size[0] * 2, originial_size[1]))
+
+        # Scaling on y-axis only
+        actor.scale_y = 2
+        self.assertEqual(actor.size, (originial_size[0], originial_size[1] * 2))
+
+        # Scaling down
+        actor.scale = (.5, .5)
+        self.assertEqual(actor.size, (originial_size[0]/2, originial_size[1]/2))
+
+        # Test scale getters
+        self.assertEqual(actor.scale, (actor.scale_x, actor.scale_y))
+
+        # Rotate and then scale
+        actor.angle = 90
+        actor.scale = (.5, .5)
+        self.assertEqual(actor.angle, 90)
+        self.assertEqual((actor.width, actor.height), (originial_size[1]/2, originial_size[0]/2))
+        self.assertEqual(actor.topleft, (-13, 13))
+
+        # Test rasing exception for invalid scale parameters
+        with self.assertRaises(InvalidScaleException) as cm:
+            actor.scale = (0, -2)
+        self.assertEqual(cm.exception.args[0], 'Invalid scale values. They should be not equal to 0.')
+
+        # Test horizontal flip
+        actor.angle = 0
+        orig = images.load('alien')
+        exp = pygame.transform.flip(orig, True, False)
+        actor.scale = (-1, 1)
+        self.assertImagesEqual(exp, actor._surf)

```

Comment:
A tip is "one assert per test". This means that if a test fails, all of the other tests still run, which gives more information, when refactoring, about what you might have broken.

Reply:
http://programmaticallyspeaking.com/one-assertion-per-test-please.html

Reply:
done